packages/api/src/routes/tasks.ts

```diff
@@ -6,17 +6,31 @@ import {
   IOptionValue,
 } from "../models/battery";
 import HttpError from "../types/errors";
-import { Study } from "../models";
+import isAuthenticated from "../middleware/auth";
+import { HydratedDocument } from "mongoose";
+import { IStudy, Study, IUser } from "@/models";
 
 const router = Router();
 
-router.get("/", async (req, res, next) => {
+router.get("/", isAuthenticated, async (req, res, next) => {
   Battery.find()
     .then((batteries) => res.json(batteries))
     .catch(next);
 });
 
-router.get("/custom/:id", (req, res, next) => {
+router.get("/custom/:id", isAuthenticated, async (req, res, next) => {
+  const user = req.user as HydratedDocument<IUser>;
+  const myStudies = (await user.populate<{ studies: IStudy[] }>("studies"))
+    .studies;
+
+  if (
+    !myStudies.some((s) =>
+      s.batteries.some((b) => b.toString() === req.params["id"])
```

Comment:
could probably make all of this one query instead of populating every study - not that familiar with mongoose but i think you can do something like `Study.exists({owner: user._id, batteries: { $contains: req.params["id"] } })`